src/dmd/dinterpret.d

```diff
@@ -703,6 +715,19 @@ private Expression interpretFunction(UnionExp* pue, FuncDeclaration fd, InterSta
         e = CTFEExp.cantexp;
     }
 
+    // destroy temporaries
+    // https://issues.dlang.org/show_bug.cgi?id=22536
+    foreach(dtor; dtors)
+    {
+        auto res = interpret(dtor, istate);
+        if (auto ctfeError = res.isThrownExceptionExp())
+        {
+            ctfeError.generateUncaughtError();
+            e = CTFEExp.cantexp;
+            break;
+        }
```

Comment:
This should also propagate `CantExp`:

```diff
+        if (CTFEExp.isCantExp(res))
+        {
+            e = res;
+            break;
+        }
```

Otherwise CTFE will issue multiple errors in this example:

```d
struct S
{
    static int global;

    ~this()
    {
        global++;
    }
}

void both(scope S[] a, scope S[] b) {}
bool test()
{
    both([ S() ], [ S() ]);
    return true;
}
static assert(test());
```